api/api/serializers/media_serializers.py

```diff
@@ -486,6 +486,34 @@ def validate(self, data):
 
         return data
 
+    def public_cache_control_allowed(self):
+        """
+        Determine whether a search request's params allow public caching of the result.
+
+        Public caches (e.g., Cloudflare) by default ignore requests with authentication
+        headers. That makes sense for routes where the returned data is personalised
+        for the authenticated user. However, when there is overlap between the possible
+        results for authenticated and unauthenticated users, in the cases where the result
+        would be the same, we can set `public` on the cache-control headers to advise
+        public caches that it is safe to cache the response, despite the authentication header.
+
+        For Openverse search, any search request that does not utilise auth-only features
+        can be publicly cached.
+        """
+        if self.is_request_anonymous():
+            return True
+
+        validated_data = self.validated_data
+        # Check if any auth-only features are used.
+        # Only allow public cache control if
+        return (
```

Comment:
Might be more readable if we extract this as a function, something like `request_includes_auth_only_features`